pkgs/development/libraries/science/math/suitesparse/default.nix

```diff
@@ -41,8 +44,6 @@ stdenv.mkDerivation rec {
     "INSTALL=${placeholder "out"}"
     "INSTALL_INCLUDE=${placeholder "dev"}/include"
     "JOBS=$(NIX_BUILD_CORES)"
-    "BLAS=-lblas"
-    "LAPACK=-llapack"
     "MY_METIS_LIB=-lmetis"
   ] ++ stdenv.lib.optionals blas.isILP64 [
```

Comment:
I'm not sure I understand the idea. `blas` is evaluated here as well.

Reply:
I think that on Darwin it will try to link against `Accelerate` unless `BLAS`/`LAPACK` are set.

https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/1869379f464f0f8dac471edb4e6d010b2b0e639d/SuiteSparse_config/SuiteSparse_config.mk#L368

Reply:
Good Catch :). We could also use darwin's `Accelerate` I guess? What is the reason we prefer  our blas && lapack against `Accelerate`?

Reply:
The goal of the new alternatives mechanism is to be able to set a BLAS/LAPACK implementation system-wide. If we'd use `Accelerate` in specific derivations, then it would not be possible anymore to override them anymore with the same mechanism for those derivations. Being able to override BLAS/LAPACK is very useful, even on Darwin, because MKL generally beats them all (also OpenBLAS is incompatible with multithreaded applications).

I guess the BLAS/LAPACK provider for Darwin could be set to Accelerate. But I am not sure if this is currently possible, since the package should provide the right library names/symlinks. Also, I don't think Accelerate currently implements LAPACKE, so you'd need a library that implements the LAPACKE interface on top of Accelerate.

Reply:
@danieldk the current PR uses `BLAS=-lblas` and `LAPACK=-llapack` as requested.